src/lib/widget/_utilities/api.utilities.ts

```diff
@@ -1,254 +1,213 @@
+      set(options: widgetConfigOptions): void {
+        configure({
+          // Set some basics by default
+          ...{
+            // TODO: Could this be a default OAuth client provided by Platform UI OOTB?
+            clientId: 'WebLoginWidgetClient',
+            // TODO: If a realmPath is not provided, should we call the realm endpoint and detect a likely default?
+            // https://backstage.forgerock.com/docs/am/7/setup-guide/sec-rest-realm-rest.html#rest-api-list-realm
+            realmPath: 'alpha',
+            // TODO: Once we move to SSR, this default should be more intelligent
+            redirectUri:
+              typeof window === 'object' ? window.location.href : 'https://localhost:3000/callback',
+            scope: 'openid email',
+          },
+          // Let user provided config override defaults
+          ...options.config,
+          // Force 'legacy' to remove confusion
+          ...{ support: 'legacy' },
+        });
+
         /**
-         * Clean up unneeded subscription, but only when it's successful
-         * Leaving the subscription allows for the journey to be
-         * restarted internally.
+         * Initialize the stores and ensure both variables point to the same reference.
+         * Variables with _ are the reactive version of the original variable from above.
          */
-        if (response.successful) {
-          journeyStoreUnsub();
-        }
-      });
+        journeyStore = initializeJourney(options.config);
+        oauthStore = initializeOauth(options.config);
+        userStore = initializeUser(options.config);
+
+        initializeContent(options.content);
+        initializeJourneys(options.journeys);
+        initializeLinks(options.links);
+        initializeStyle(options.style);
+      },
+    };
+  };
+  const journey = (options?: JourneyOptions) => {
+    const requestsOauth = options?.oauth || true;
+    const requestsUser = options?.user || true;
+    const { subscribe } = derived(
+      [journeyStore, oauthStore, userStore],
+      ([$journeyStore, $oauthStore, $userStore], set) => {
+        set({
+          journey: $journeyStore,
+          oauth: $oauthStore,
+          user: $userStore,
+        });
 
-      const oauthStoreUnsub = oauthStore.subscribe((response) => {
-        if (!requestsUser && response.successful) {
-          returnResponse &&
-            returnResponse({
-              journey,
-              oauth: response,
-            });
-          if (modal) {
-            modal.close({ reason: 'auto' });
+        if ($journeyStore.successful && $oauthStore.successful && $userStore.completed) {
+          modal && modal.close({ reason: 'auto' });
+        } else if ($journeyStore.successful && $oauthStore.successful) {
+          if (requestsUser && $userStore.loading === false && $userStore.completed === false) {
+            userStore.get();
+          } else if (!requestsUser) {
+            modal && modal.close({ reason: 'auto' });
```

Comment:
Much of this logic is just a simplified version of what already existed. Using a derived store was much easier because I didn't have to subscribe and unsubscribe to each store as the widget progressed through the auth flow.